feature-libs/user/account/components/login-form/login-form-component.service.ts

```diff
@@ -40,35 +58,60 @@ export class LoginFormComponentService {
     })
   );
 
-  form: UntypedFormGroup = new UntypedFormGroup({
-    userId: new UntypedFormControl('', [
-      Validators.required,
-      CustomFormValidators.emailValidator,
-    ]),
-    password: new UntypedFormControl('', Validators.required),
-  });
+  form: UntypedFormGroup = this.isNewAuthFlow
+    ? new FormGroup({
+        userId: new FormControl('', [
+          Validators.required,
+          CustomFormValidators.emailValidator,
+        ]),
+        password: new FormControl('', Validators.required),
+        csrf: new FormControl('', Validators.required),
+      })
+    : new UntypedFormGroup({
+        userId: new UntypedFormControl('', [
+          Validators.required,
+          CustomFormValidators.emailValidator,
+        ]),
+        password: new UntypedFormControl('', Validators.required),
+      });
 
-  login() {
-    if (!this.form.valid) {
-      this.form.markAllAsTouched();
-      return;
-    }
+  login(nativeForm?: HTMLFormElement) {
+    if (this.isNewAuthFlow && nativeForm) {
+      const csrf =
+        [...nativeForm.elements]
+          .find((element) => element.hasAttribute?.('data-csrf'))
+          ?.getAttribute?.('data-csrf') ?? '';
+      this.form.get('csrf')?.setValue(csrf);
```

Comment:
`query`: what's the purpose of this logic? If I'm not mistaken, the csrf is already part of the form and should have access to the value. Apart from that, the nativeForm do the job here, so I'm missing the reason of explicitly setting the value to the form control.